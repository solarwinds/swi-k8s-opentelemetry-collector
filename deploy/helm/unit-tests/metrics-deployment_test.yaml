# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for metrics-deployment
templates:
  - metrics-deployment.yaml
  - metrics-collector-config-map.yaml
  - common-env-config-map.yaml
  - metrics-collector-env-config-map.yaml
chart:
  appVersion: 1.0.0
tests:
  - it: Metrics collector spec should match snapshot when using default values
    template: metrics-deployment.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: Image should be correct in default state
    template: metrics-deployment.yaml
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swi-opentelemetry-collector:1.0.0
  - it: Image should be correct when overridden repository
    template: metrics-deployment.yaml
    set:
      otel.image.repository: "swi-opentelemetry-collector"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: swi-opentelemetry-collector:1.0.0
  - it: Image should be correct when overridden tag
    template: metrics-deployment.yaml
    set:
      otel.image.tag: "beta1"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swi-opentelemetry-collector:beta1
  - it: Image should be correct when overridden by azure
    template: metrics-deployment.yaml
    set:
      aks: true
      global.azure.images.image.digest: "abcd"
      global.azure.images.image.image: "swi-opentelemetry-collector:v1.2.3"
      global.azure.images.image.registry: "azurek8s.azure.io/marketplaceimages"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: azurek8s.azure.io/marketplaceimages/swi-opentelemetry-collector:v1.2.3@abcd
    - isSubset:
        path: spec.template.metadata.labels
        content:
          azure-extensions-usage-release-identifier: RELEASE-NAME
  - it: SWI endpoint check should not be included when disabled
    template: metrics-deployment.yaml
    set:
      otel.swi_endpoint_check.enabled: false
    asserts:
    - notExists:
        path: spec.template.spec.initContainers
  - it: SWI endpoint check should be included in default state
    template: metrics-deployment.yaml
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].name
          value: otel-endpoint-check
  - it: SWI endpoint check should not be included when disabled by deprecated setting
    template: metrics-deployment.yaml
    set:
      otel.metrics.swi_endpoint_check: false
    asserts:
    - notExists:
        path: spec.template.spec.initContainers
  - it: Image otel-endpoint-check should be correct in default state
    template: metrics-deployment.yaml
    set:
      otel.swi_endpoint_check.enabled: true
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: solarwinds/swi-opentelemetry-collector:1.0.0
  - it: Image otel-endpoint-check should be correct when overridden repository
    template: metrics-deployment.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      otel.image.repository: "swi-opentelemetry-collector"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: swi-opentelemetry-collector:1.0.0
  - it: Image otel-endpoint-check should be correct when overridden tag
    template: metrics-deployment.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      otel.image.tag: "beta1"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: solarwinds/swi-opentelemetry-collector:beta1
  - it: Image otel-endpoint-check should be correct when overridden by azure
    template: metrics-deployment.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      aks: true
      global.azure.images.image.digest: "abcd"
      global.azure.images.image.image: "swi-opentelemetry-collector:v1.2.3"
      global.azure.images.image.registry: "azurek8s.azure.io/marketplaceimages"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: azurek8s.azure.io/marketplaceimages/swi-opentelemetry-collector:v1.2.3@abcd
  - it: Image prometheus-check should be correct in default state
    template: metrics-deployment.yaml
    set:
      otel.metrics.prometheus_check: true
      otel.metrics.prometheus.url: "http://prometheus:9090"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: busybox:1.36.1