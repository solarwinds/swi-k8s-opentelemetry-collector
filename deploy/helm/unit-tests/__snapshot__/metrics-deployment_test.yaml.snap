Metrics collector spec should match snapshot when using default values:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: kubernetes.io/arch
                  operator: In
                  values:
                    - amd64
                    - arm64
                - key: kubernetes.io/os
                  operator: In
                  values:
                    - linux
    containers:
      - command:
          - /swi-otelcol
          - --config=/conf/relay.yaml
        env:
          - name: MY_POD_IP
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: status.podIP
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
          - name: GOMEMLIMIT
            valueFrom:
              resourceFieldRef:
                resource: limits.memory
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-metrics-env-config
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
          initialDelaySeconds: 10
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 4317
            name: otlp
            protocol: TCP
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
          initialDelaySeconds: 10
        resources:
          limits:
            memory: 3Gi
          requests:
            memory: 3Gi
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
              - ALL
          readOnlyRootFilesystem: true
          runAsNonRoot: true
          seccompProfile:
            type: RuntimeDefault
        volumeMounts:
          - mountPath: /conf
            name: opentelemetry-collector-configmap
            readOnly: true
    initContainers:
      - command:
          - /swi-otelcol
          - test-connection
          - --endpoint=$(OTEL_ENVOY_ADDRESS)
          - --apitoken=$(SOLARWINDS_API_TOKEN)
          - --clusteruid=$(CLUSTER_UID)
          - --insecure=$(OTEL_ENVOY_ADDRESS_TLS_INSECURE)
        env:
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        name: otel-endpoint-check
    securityContext: {}
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 600
    volumes:
      - configMap:
          items:
            - key: metrics.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-metrics-config
        name: opentelemetry-collector-configmap
